Laser fixes

PYROTEKNIK opened the pull request:
some fixes that don't appear to have made it on

Commit:
Redone all changes i guess

i fucked something up bad so here's everything again?

Commit:
Update weapon_laserpointer.lua

right click sound filtering so that it can't click out unless it's already been clicked in

Commit:
Update weapon_laserpointer.lua

fov_desired seems to fuck up so i'm using the fov input instead . It disconnects while the FOV is changing via suit zoom but it's better than randomly setting to the wrong FOV forever

Commit:
Update sv_vapeswep.lua

rare chance for vape to explode when exhaling

Commit:
Update weapon_laserpointer.lua

replaced drawsprite with drawquadeasy. beam ends now have random rotation for a flickery laser effect.
changed holdtype to knife

Commit:
Update weapon_laserpointer.lua

yoinked the hack for the vape to allow it to render on ponies

Commit:
Update weapon_laserpointer.lua

oops my hook wasn't being called correctly.

Commit:
Update weapon_laserpointer.lua

more visual tweaks

Commit:
Update weapon_laserpointer.lua

Requested changes, reformat.

Commit:
Update weapon_laserpointer.lua

holy shit hahaha what is this formatting

Commit:
Update weapon_laserpointer.lua

laser should pass through player clips

Commit:
Laserpointer Store Fix

category adjustment and store addition for laser pointer

Commit:
added fake ammotype to override multibuy protection

added fake ammotype to override multibuy protection

Commit:
Update weapon_laserpointer.lua

bounces off glass surfaces

Commit:
Update weapon_laserpointer.lua

correct masking

Commit:
Update weapon_laserpointer.lua

more consistent pointer length

Commit:
Update shared.lua

louder braps

Commit:
Merge branch 'master' into master

Commit:
Update weapon_laserpointer.lua

dissolve enemies on kill

Commit:
Merge branch 'master' of https://github.com/PYROTEKNIK/contrib

GrateBrittan commented on lua/sps/items/other.lua:
```diff
@@ -58,6 +58,15 @@ PS_WeaponProduct({
+PS_WeaponProduct({
+	name = 'Laser Pointer',
+	description = "Makes a funny dot. Keep away from eyes. Right click for lethal beam. Re-buy for battery refill.",
+	price = 350,
```
Price should be higher, current price is wayyy to overpowered. This is cheapest weapon on the shop, I think it should be like 1000 or more.

swampservers replied:
Isn't it already buyable through console and therefore already being registered somewhere? also yeah too cheap

PYROTEKNIK replied:
I moved it to the file that contained other items in the expensive toys category. Also I feel the price is justified since its capacity to kill is basically just above nothing. One battery has the potential to do exactly 250 damage to one person at a time over a period of 50 seconds. Please provide a price suggestion if you do not agree with mine.

Commit:
Update weapon_laserpointer.lua

This might be a little fucked. The formatting site you sent me replaced all of the indenting with spaces. Also R now just sets a custom color.

Commit:
Update weapons.lua

whoops duplicate entry wasn't removed from weapons

Commit:
Update other.lua

adjusted price to 500
